### PR TITLE
Add Java datastore process.

### LINKED ISSUE #30: Aggregate data uploaded from reporters and cut histogram tiles
- using the agreed upon file naming and format found in https://github.com/opentraffic/reporter/issues/70
- using the histogram tile format defined for #28 (based on research in https://github.com/opentraffic/histogram-format-tests)

### PR DESCRIPTION
**Pull request by zerebubuth**
closes #30 

---

The process reads one or more input files from the local file system or S3, processes and buckets them, and writes the resulting data to a FlatBuffers file or an ORC file, again in either S3 or the local filesystem.

The intention is that this is called one or more times for each time a file is uploaded which covers a `(tile, hour)` tuple that is wanted in the output.

### CODE REVIEW

**Comment by kevinkreiser** [comment]
cool so ill rework the circle tests by providing a sample output from the reporter stuff im working on and seeing that this indeed transforms it as expected. at any rate ill give a proper review as well on monday

File: src/main/fbs/histogram-tile.fbs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  // note: could be 25 to account for the single 25th hour when DST changes?
+  day_hour:ubyte;
+
+  // index into the next_segment_ids array in the Segment.
+  // note: imposes a limit of 256 segments leaving any one segment.
+  next_segment_idx:ubyte;
+
+  // bucket in 5mph intervals.
+  // note: do we set a max speed bucket, e.g: 120mph?
+  speed_bucket:ubyte;
```
Comment (dnesbitt61) [inline src/main/fbs/histogram-tile.fbs]: is this right? weren't these now time / duration buckets?

Comment (zerebubuth) [reply src/main/fbs/histogram-tile.fbs]: Thanks for spotting that! I had forgotten to change that, but fixed in 2939adf.

I had been using it as a duration bucket throughout the code, but forgot to change the name of the field in the FB definition.

**Comment by kevinkreiser** [comment]
ive merged my changes into this pr. need to update the tests to check the `-v` output of the program

**Review by kevinkreiser — APPROVED** [review]
we've changed this around a bit and still need to add the bit where it can read back in flatbuffer to combine previously made histograms. we'll tackle that in another pr

**Pull request merged**